sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -680,48 +667,37 @@ private void validateDBBody(DocumentBody body) {
     }
 
     @Override
-    public BasicDocumentRevision createLocalDocument(final DocumentBody body) throws DocumentException {
-        Preconditions.checkState(this.isOpen(), "Database is closed");
-        String documentId = CouchUtils.generateDocumentId();
-        return createLocalDocument(documentId, body);
-    }
-
-    @Override
-    public BasicDocumentRevision createLocalDocument(final String docId, final DocumentBody body)
-            throws DocumentException {
+    public LocalDocument insertLocalDocument(final String docId, final DocumentBody body) throws DocumentException {
         Preconditions.checkState(this.isOpen(), "Database is closed");
         CouchUtils.validateDocumentId(docId);
         Preconditions.checkNotNull(body, "Input document body can not be null");
         try {
-            return queue.submitTransaction(new SQLQueueCallable<BasicDocumentRevision>(){
+            return queue.submitTransaction(new SQLQueueCallable<LocalDocument>() {
                 @Override
-                public BasicDocumentRevision call(SQLDatabase db) throws Exception {
-                        String firstRevId = CouchUtils.getFirstLocalDocRevisionId();
-                        ContentValues values = new ContentValues();
-                        values.put("docid", docId);
-                        values.put("revid", firstRevId);
-                        values.put("json", body.asBytes());
-
-                        long lineId = db.insert("localdocs", values);
-                        if (lineId < 0) {
-                            throw new DocumentException("Can not insert new local doc, likely the docId exists already: "
-                                    + docId);
-                        } else {
-                            logger.finer("New local doc inserted: " + lineId + ", " + docId);
-                        }
+                public LocalDocument call(SQLDatabase db) throws Exception {
+                    ContentValues values = new ContentValues();
+                    values.put("docid", docId);
+                    values.put("json", body.asBytes());
+
+                    long rowId = db.insertWithOnConflict("localdocs", values, SQLDatabase
+                            .CONFLICT_REPLACE);
+                    if (rowId < 0) {
+                        throw new DocumentException("Failed to insert local document");
```

Comment:
These log messages and exception messages here and below are a bit varied: "failed/cannot", "create/insert" - they should be consistent.